ext/com/concurrent_ruby/ext/JavaSemaphoreLibrary.java

```diff
@@ -0,0 +1,147 @@
+package com.concurrent_ruby.ext;
+
+
+import java.io.IOException;
+import java.util.concurrent.Semaphore;
+import org.jruby.Ruby;
+import org.jruby.RubyBoolean;
+import org.jruby.RubyClass;
+import org.jruby.RubyFixnum;
+import org.jruby.RubyModule;
+import org.jruby.RubyNumeric;
+import org.jruby.RubyObject;
+import org.jruby.anno.JRubyClass;
+import org.jruby.anno.JRubyMethod;
+import org.jruby.runtime.ObjectAllocator;
+import org.jruby.runtime.ThreadContext;
+import org.jruby.runtime.builtin.IRubyObject;
+
+public class JavaSemaphoreLibrary {
+
+    public void load(Ruby runtime, boolean wrap) throws IOException {
+        RubyModule concurrentMod = runtime.defineModule("Concurrent");
+        RubyClass atomicCls = concurrentMod.defineClassUnder("JavaSemaphore", runtime.getObject(), JRUBYREFERENCE_ALLOCATOR);
+
+        atomicCls.defineAnnotatedMethods(JavaSemaphore.class);
+
+    }
+
+    private static final ObjectAllocator JRUBYREFERENCE_ALLOCATOR = new ObjectAllocator() {
+        public IRubyObject allocate(Ruby runtime, RubyClass klazz) {
+            return new JavaSemaphore(runtime, klazz);
+        }
+    };
+
+    @JRubyClass(name = "JavaSemaphore", parent = "Object")
+    public static class JavaSemaphore extends RubyObject {
+
+        private JRubySemaphore semaphore;
+        private ThreadContext context;
```

Comment:
@pitr-ch Makes completely sense! I'll remove it here and I'll open a new pull request to remove the `ThreadContext` from `JavaAtomicFixnum` and `JavaBooleanFixnum`.